internal/database/postgres.go

```diff
@@ -309,66 +311,125 @@ FROM TempDownsampled;
 }
 
 // TODO: consider returning workstationGroup
-func (conn PostgresConn) LatestData() ([]uplink.GpuStatsUpload, error) {
-	// we pull Uuid twice so we can put one into Stat sample and the other into Info
-	rows, err := conn.db.Query(`SELECT g.Machine, g.Uuid, g.Uuid, g.Name,
-			g.Brand, g.DriverVersion, g.MemoryTotal,
-			s.MemoryUtilisation, s.GpuUtilisation, s.MemoryUsed,
-			s.FanSpeed, s.Temp, s.MemoryTemp, s.GraphicsVoltage,
-			s.PowerDraw, s.GraphicsClock, s.MaxGraphicsClock,
-			s.MemoryClock, s.MaxMemoryClock
+func (conn PostgresConn) LatestData() (broadcast.Workstations, error) {
```

Comment:
Isn't it possible here to use SQL's grouping mechanism to group things for you. SQL engines are **heavily** optimised for this work, why are we reinventing the wheel?

Reply:
I don't think a `GROUP BY` query, if that's what you're suggesting, would work for `LatestData`, because we don't want to aggregate the GPUs so there's one row per machine/per group